target/linux/generic/pending-6.12/811-pci_disable_usb_common_quirks.patch

```diff
@@ -20,16 +20,25 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  static struct amd_chipset_info {
  	struct pci_dev	*nb_dev;
  	struct pci_dev	*smbus_dev;
-@@ -590,6 +592,8 @@ bool usb_amd_pt_check_port(struct device
+@@ -590,6 +592,7 @@ bool usb_amd_pt_check_port(struct device
  EXPORT_SYMBOL_GPL(usb_amd_pt_check_port);
  #endif /* CONFIG_USB_PCI_AMD */
  
-+#endif /* CONFIG_PCI_DISABLE_COMMON_QUIRKS */
-+
++#ifdef CONFIG_USB_PCI
  static int usb_asmedia_wait_write(struct pci_dev *pdev)
  {
  	unsigned long retry_count;
-@@ -724,6 +728,10 @@ reset_needed:
+@@ -633,6 +636,9 @@ void usb_asmedia_modifyflowcontrol(struc
+ 	pci_write_config_byte(pdev, ASMT_CONTROL_REG, ASMT_CONTROL_WRITE_BIT);
+ }
+ EXPORT_SYMBOL_GPL(usb_asmedia_modifyflowcontrol);
++#endif /* CONFIG_USB_PCI */
++
++#endif /* CONFIG_PCI_DISABLE_COMMON_QUIRKS */
+ 
+ static inline int io_type_enabled(struct pci_dev *pdev, unsigned int mask)
+ {
+@@ -724,6 +730,10 @@ reset_needed:
  	return 1;
  }
  EXPORT_SYMBOL_GPL(uhci_check_and_reset_hc);
```

Comment:
The original patch exports an empty version of `uhci_check_and_reset_hc` when `#if IS_ENABLED(CONFIG_USB_UHCI_HCD)` is false.

Reply:
In 6.12 kernel upstream commit https://github.com/torvalds/linux/commit/358ad297e379ff548247e3e24c6619559942bfdd, `uhci_check_and_reset_hc` and `quirk_usb_handoff_uhci` are guarded by `#if IS_ENABLED(CONFIG_USB_UHCI_HCD)`. Hence `uhci_check_and_reset_hc()` is **unused** if not define `CONFIG_USB_UHCI_HCD`. We don't need to hack it again.

Reply:
ping @Ansuel